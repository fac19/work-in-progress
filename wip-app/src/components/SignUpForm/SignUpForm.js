import React from "react";
import { Container, Button } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import { useHistory } from "react-router-dom";
import { signUpPost } from "../../utils/post-fetch";

const useStyles = makeStyles({
  form: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    width: "90%",
  },
  formContainer: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
  },
  formElement: {
    margin: "0.5rem",
  },
});

const SignUpForm = (props) => {
  const classes = useStyles();
  const history = useHistory();

  const handleSubmit = (event) => {
    event.preventDefault();
    const form = document.querySelector("form");
    const formData = new FormData(form);

    signUpPost({
      username: formData.get("username"),
      email: formData.get("email"),
      password: formData.get("password"),
    })
      .then(() => history.push("/feed"))
      .catch((error) => console.error(error));
  };

  return (
    <Container className={classes.formContainer} component="main" maxWidth="xs">
      <h1>Sign Up</h1>
      <form className={classes.form} onSubmit={handleSubmit}>
        <label htmlFor="username">Username *</label>
        <input
          type="text"
          id="username"
          placeholder="username"
          name="username"
          required
          autoFocus
        ></input>
        <label htmlFor="email">Email *</label>
        <input
          type="email"
          id="email"
          placeholder="email"
          name="email"
          required
        ></input>
        <label htmlFor="password">Password *</label>
        <input
          type="password"
          id="password"
          name="password"
          placeholder="Password"
          required
        ></input>
        <Button
          className={classes.formElement}
          variant="contained"
          color="primary"
          type="submit"
        >
          Sign Up
        </Button>
      </form>
    </Container>
  );
};

export default SignUpForm;
