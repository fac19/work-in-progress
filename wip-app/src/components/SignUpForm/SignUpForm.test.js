import React from "react";
import TestRenderer from "react-test-renderer";
import SignUpForm from "./SignUpForm";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

describe("Snapshot test", () => {
  it("renders the correct ui", () => {
    const history = createMemoryHistory();
    const component = TestRenderer.create(
      <Router history={history}>
        <SignUpForm />
      </Router>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
