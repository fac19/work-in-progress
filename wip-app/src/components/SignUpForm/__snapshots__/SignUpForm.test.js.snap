// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders the correct ui 1`] = `
<main
  className="MuiContainer-root makeStyles-formContainer-2 MuiContainer-maxWidthXs"
>
  <img
    alt="work in progress logo"
    className="sc-AxjAm ecCdST"
    src="logo.svg"
  />
  <h1>
    Sign Up
  </h1>
  <form
    className="makeStyles-form-1"
    onSubmit={[Function]}
  >
    <label
      className="sc-AxhCb djpVYo"
      htmlFor="username"
    >
      Username *
    </label>
    <input
      autoFocus={true}
      className="sc-AxiKw dxRHDb"
      id="username"
      name="username"
      onChange={[Function]}
      placeholder="Username"
      required={true}
      type="text"
    />
    <label
      className="sc-AxhCb djpVYo"
      htmlFor="email"
    >
      Email *
    </label>
    <input
      className="sc-AxiKw dxRHDb"
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
    />
    <label
      className="sc-AxhCb djpVYo"
      htmlFor="password"
    >
      Password *
    </label>
    <input
      className="sc-AxiKw dxRHDb"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
    />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-formElement-3 MuiButton-containedSecondary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Sign Up
      </span>
    </button>
  </form>
  <p
    className="makeStyles-errorMessage-4"
  >
    
  </p>
  <p>
    If you already have an account, please
     
    <a
      className="makeStyles-link-5"
      href="/log-in"
      onClick={[Function]}
    >
      log in
    </a>
  </p>
</main>
`;
